lib/foreman_maintain/cli/upgrade_command.rb

```diff
@@ -52,12 +55,18 @@ def print_versions(target_versions)
         target_versions.sort.each { |version| puts version }
       end
 
+      def do_self_upgrade?
```

Comment:
`do_self_upgrade?` sounds (to me) like "this will tell you if we're in the middle of a self-upgrade", not "check whether self-upgrade is supported in this scenario".

`allow_self_upgrade?`, `self_upgrade_supported?`, idk what's better (feel free to ignore me, too!)